### ISSUE TITLE
Specifying ARM_SUBSCRIPTION_ID is now mandatory when using the cluster-autoscaler on AKS with MSI auth

### ISSUE THREAD

**Issue by jclangst** [kubernetes/autoscaler#4635]
**Which component are you using?**:

`cluster-autoscaler`

**What version of the component are you using?**:

`1.22.0+`

Component version:

**What k8s version are you using (`kubectl version`)?**:

`1.22.4`

<details><summary><code> kubectl version
</code> Output</summary><br><pre>
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-18T19:30:35Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}

</pre></details>

**What environment is this in?**:

`AKS`

**What did you expect to happen?**:

When using MSI authentication (`ARM_USE_MANAGED_IDENTITY_EXTENSION` = `"true"`), I expect the Azure Manager to be instantiated successfully without having to specify the subscription id (`ARM_SUBSCRIPTION_ID`) as explained in the [docs here](https://github.com/kubernetes/autoscaler/blob/5c741c881d194ba600419b79cfcf195f57b82d11/cluster-autoscaler/cloudprovider/azure/README.md?plain=1#L87).

**What happened instead?**:

The Azure Manager is not created and the cluster-autoscaler exits with this message:

`F0120 20:37:10.851256       1 azure_cloud_provider.go:167] Failed to create Azure Manager: failure of getting instance metadata with response "404 Not Found"`

This happens _prior_ to attempting to fetch credentials from the IMDS endpoint. In particular, it appears the changes introduced in `1.22.0` cause the cluster autoscaler to attempt this request:

```
GET /metadata/instance/metadata/instance?api-version=2019-03-11&format=json HTTP/1.1
Host: 169.254.169.254:80
User-Agent: golang/kubernetes-cloud-provider
Accept-Encoding: gzip
Metadata: True
```
which is an invalid address and results in the `404` found in the logs.

Providing the subscription id explicitly via `ARM_SUBSCRIPTION_ID` avoids this code path and allows the Azure Manager to be instantiated correctly.

I've confirmed that this issue began in `1.22.0` and fetching the subscription id dynamically works in `1.21.2`. The issue appears to be related to the upgrade of the azure SDK that occurred in that release. Additionally, the `1.22.2` release removes the dynamic subscription fetching code as a part of afb443f9f35a0014ef60589bceabe90e34001665 (`azure_config.go`).

I'd suggest that:

- the documentation for Azure be updated to indicate that the `ARM_SUBSCRIPTION_ID` parameter is mandatory
- a breaking change notice be added to the `1.22.0` [release notes](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.22.0) regarding the `ARM_SUBSCRIPTION_ID`

**How to reproduce it (as minimally and precisely as possible)**:

Use the cluster-autoscaler's [MSI example deployment](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/azure/examples/cluster-autoscaler-vmss-msi.yaml) without specifying `ARM_SUBSCRIPTION_ID`.

**Other notes**:

It appears the invalid instance metadata endpoint fix (`/metadata/instance/metadata/instance`) has been incorporated as a part of #4600. However, the issue with the dynamic fetching of the subscription id is still valid.

**Comment by marwanad** [comment]
The dynamic subscription fetching was removed in 1.23 and is being added back.

Can you confirm that #4600 doesn't fix your issue?

**Comment by jclangst** [comment]
I can confirm #4600 fixes the invalid metadata URL, and I can confirm that the 1.23 release does not appear to have the subscription fetching code (removed in https://github.com/kubernetes/autoscaler/commit/afb443f9f35a0014ef60589bceabe90e34001665) which was added in 1.22.2. Seems like a breaking change that should be added to the 1.22.2 release notes (or re-added for 1.22.2 and removed in 1.23 only). Either way, the docs are still incorrect in those releases as they explicitly state that the dynamic subscription fetching is supported.

**Comment by marwanad** [comment]
@jclangst 

We will fix the 1.23 branch :) The owner of that PR mentioned they're bringing the fix back and will be noted in the release notes.

**Comment by jclangst** [comment]
@marwanad Sounds like a plan. Thank you!
